File: app/src/main/java/com/milnest/tasklist/JsonAdapter.java
package com.milnest.tasklist;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

/**
 * Created by t-yar on 24.04.2018.
 * Работает с json (парсит в список и обратно)
 */

public class JsonAdapter {
    private static GsonBuilder builder = new GsonBuilder();
    private static Gson gson = builder.create();

    public static String toJson(ListOfCheckboxesTaskListItem listItem) {
        return gson.toJson(listItem);
    }


    public static ListOfCheckboxesTaskListItem  fromJson(String jsonString){
        return gson.fromJson(jsonString, ListOfCheckboxesTaskListItem.class);
    }
}

File: app/src/main/java/com/milnest/tasklist/JsonAdapter.kt
package com.milnest.tasklist

import com.google.gson.Gson
import com.google.gson.GsonBuilder

/**
 * Created by t-yar on 24.04.2018.
 * Работает с json (парсит в список и обратно)
 */

object JsonAdapter {
    private val builder = GsonBuilder()
    private val gson = builder.create()

    fun toJson(listItem: ListOfCheckboxesTaskListItem): String {
        return gson.toJson(listItem)
    }


    fun fromJson(jsonString: String): ListOfCheckboxesTaskListItem {
        return gson.fromJson(jsonString, ListOfCheckboxesTaskListItem::class.java!!)
    }
}
